src/NHibernate/Type/ManyToOneType.cs

```diff
@@ -144,6 +157,18 @@ private bool IsIdentifier(object value, ISessionImplementor session)
 			return value.GetType() == identifierType.ReturnedClass;
 		}
 
+		public override bool IsNull(object owner, ISessionImplementor session)
+		{
+			if (IsNullable && !string.IsNullOrEmpty(PropertyName))
+			{
+				EntityEntry entry = session.PersistenceContext.GetEntry(owner);
+
+				return session.PersistenceContext.IsPropertyNull(entry.EntityKey, PropertyName);
```

Comment:
This causes #2627 regression. During a merge, the owner does not belong to the session, causing `GetEntry` to yield `null`.

The `OneToOne` type has a different implementation which does not rely on the owner belonging to session, using a list of known `null` associations, which this PR seems to update with `not-found` cases. So why not having used the `OneToOne` implementation here?

https://github.com/nhibernate/nhibernate-core/blob/33f90d3742f1a2d5a2aa2368edbaf667663fae1a/src/NHibernate/Type/OneToOneType.cs#L77-L85